### PR TITLE
Feat/keccak in zkc

### PR DESCRIPTION
**Pull request by amkCha**
> [!NOTE]
> **Low Risk**
> Changes are limited to adding new ZKC benchmark/unit test coverage and associated testdata; no production logic is modified. Main risk is increased CI runtime or flaky failures if the new Keccak/rotation fixtures expose backend edge cases.
> 
> **Overview**
> Adds a new ZKC benchmark test for `zkc/bench/keccakf` and parks (commented) placeholders for future Keccak variants.
> 
> Adds a new ZKC unit test `zkc/unit/bit_rotr64` with `.zkc` program and `.accepts` vectors, and introduces Keccak-f benchmark programs in `testdata/zkc/bench` (big-endian, little-endian, and padding variants) to support current/future benchmarks.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit fe9ae287afcb0d5d78ec0a17ee4892a2e5a3e6ff. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: testdata/zkc/bench/keccakf_le.zkc [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+        state_row1[j] = bit_rol64(state_row1[j], rot_offset(1, j))
+        state_row2[j] = bit_rol64(state_row2[j], rot_offset(2, j))
+        state_row3[j] = bit_rol64(state_row3[j], rot_offset(3, j))
+        state_row4[j] = bit_rol64(state_row4[j], rot_offset(4, j))
+    }
+}
+
+// PI: lane permutation step
+// A[x,y] → B[y, (2x+3y) mod 5]
+fn pi<state_row0, state_row1, state_row2, state_row3, state_row4, b0, b1, b2, b3, b4>() {
```
Comment (OlivierBBB) [inline testdata/zkc/bench/keccakf_le.zkc:194]: Silly remark(?) but all of these `mod5((2 * x) + (3 * y))` are static constants. They don't warrant a lookup to the `mod5` static table `zkc` which will amount to a lookup on the constraints side, too. Also the computation of `(2 * x) + (3 * y)` itself will generate registers to hold various intermediate results. I don't think `go-corset` is smart enough today to not create associated columns downstream.

I would suggest replacing them with their explicit values (e.g. `mod5((2 * 4) + (3 * 1)) -> 1`)and leaving the computation as a comment à la
```
bY[Z] = state_rowX[Y] with Z = (2*X + 3*Y) mod 5
```
or so.

File: testdata/zkc/bench/keccakf.zkc [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+// RHO: lane rotations step
+// B[y, (2x+3y) mod 5] = rot(A[x,y], r[x,y])
+fn rho<state_row0, state_row1, state_row2, state_row3, state_row4>() {
+
+    for j:u3 = 0; j<5; j = j + 1 {
+        state_row0[j] = bit_rol64(state_row0[j], rot_offset(0, j))
+        state_row1[j] = bit_rol64(state_row1[j], rot_offset(1, j))
+        state_row2[j] = bit_rol64(state_row2[j], rot_offset(2, j))
+        state_row3[j] = bit_rol64(state_row3[j], rot_offset(3, j))
+        state_row4[j] = bit_rol64(state_row4[j], rot_offset(4, j))
```
Comment (OlivierBBB) [inline testdata/zkc/bench/keccakf.zkc]: Micro comment: the `bit_rol` stands for `bit rotate left` from what I gather and on the right you use `rot` as the diminutive of rotation. Maybe we can rename `rol` to `rotl` ? Same comment applies to `ror`.

File: testdata/zkc/bench/keccakf_le.zkc [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+
+    c[0] = state_row0[0] ^ state_row0[1] ^ state_row0[2] ^ state_row0[3] ^ state_row0[4]
+    c[1] = state_row1[0] ^ state_row1[1] ^ state_row1[2] ^ state_row1[3] ^ state_row1[4]
+    c[2] = state_row2[0] ^ state_row2[1] ^ state_row2[2] ^ state_row2[3] ^ state_row2[4]
+    c[3] = state_row3[0] ^ state_row3[1] ^ state_row3[2] ^ state_row3[3] ^ state_row3[4]
+    c[4] = state_row4[0] ^ state_row4[1] ^ state_row4[2] ^ state_row4[3] ^ state_row4[4]
+
+    for i_minus_one:u3 = 0; i_minus_one<5; i_minus_one = i_minus_one + 1 {
+        var i:u3 = i_minus_one + 1
+        d[mod5(i as u5)] = c[i_minus_one] ^ bit_rol64(c[mod5((i + 1) as u5)], 1)
```
Comment (OlivierBBB) [inline testdata/zkc/bench/keccakf_le.zkc]: Maybe silly remark but you have two casts `u3 -> u5` when you could make `i_minus_one` a `u5` and only have one (safe) cast `u5 -> u3`. C'est epsilonesque, I agree.

**Review by OlivierBBB — APPROVED** [review]
LGTM, some minor comments, one which suggests replacing the `mod5((2 * x) + (3 * y))` with their explicit values may have a slight impact on the number of registers / columns produced, see https://github.com/Consensys/go-corset/issues/1724.

File: testdata/zkc/bench/keccakf_le.zkc [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+// ===================================================================
+
+// State is a 5x5xW matrix
+// For a state size of 1600, W = 64 (5x5x64 = 1600 bits)
+// State is a 5x5 matrix with 64-bit lanes
+type lane = u64
+// The output is 256 bits, so capacity is 2 * 256 = 512 bits
+// Rate = State size - Capacity = 1600 - 512 = 1088 bits
+const bitrate:u64 = 1088
+type bitrate_type = u1088
```
Comment (OlivierBBB) [inline testdata/zkc/bench/keccakf_le.zkc:15]: Actually, is it wise to have such large integers ? You'll have many constraints / columns associated with destructuring.

File: testdata/zkc/bench/keccakf_with_padding.zkc [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+// Keccak-f[1600] parameters and constants
+// ===================================================================
+
+// State is a 5x5xW matrix
+// For a state size of 1600, W = 64 (5x5x64 = 1600 bits)
+// State is a 5x5 matrix with 64-bit lanes
+type lane = u64
+// The output is 256 bits, so capacity is 2 * 256 = 512 bits
+// Rate = State size - Capacity = 1600 - 512 = 1088 bits
+const bitrate:u64 = 1088
```
Comment (OlivierBBB) [inline testdata/zkc/bench/keccakf_with_padding.zkc:15]: Is that constant ever used ?

Comment (amkCha) [reply testdata/zkc/bench/keccakf_with_padding.zkc:15]: Actually you're right, `bitrate` is only used in `keccakf_with_padding`, not in the 2 other keccakf. It's used in the different steps in order to do padding, to know when to move on to the next block. For instance : 

```
    // If message is larger than one block, fill complete blocks
    while ((index + 1) * bitrate)<message_length {
        blocks[index] = extract_bitrate_from_msg(raw_message, message_start + (index * bitrate))
        index = index + 1
    }
```

File: testdata/zkc/bench/keccakf_le.zkc (hunk 4 above)
Comment (amkCha) [reply testdata/zkc/bench/keccakf_le.zkc:15]: I see your point. It's the size of a block, which I then split into u64. So another program could pre-chop the blocks into lanes

File: testdata/zkc/bench/keccakf_le.zkc (hunk 3 above)
Comment (amkCha) [reply testdata/zkc/bench/keccakf_le.zkc]: done :)

File: testdata/zkc/bench/keccakf_le.zkc (hunk 1 above)
Comment (amkCha) [reply testdata/zkc/bench/keccakf_le.zkc:194]: I didn't have the constraints view in mind - I had tried that to see if it improved the benchmarks and it didn't so left it, but there were also the constraints (!)

Comment (amkCha) [reply testdata/zkc/bench/keccakf_le.zkc:194]: updated :ok_hand:

File: testdata/zkc/bench/keccakf.zkc (hunk 2 above)
Comment (amkCha) [reply testdata/zkc/bench/keccakf.zkc]: changed

File: testdata/zkc/bench/keccakf_le.zkc (hunk 4 above)
Comment (OlivierBBB) [reply testdata/zkc/bench/keccakf_le.zkc:15]: Ultimately inputs and thus blocks will come from the RV program's ram, which is
```
memory ram(address :Address) -> (byte :u8)
```
I don't know what the best solution for blocks is then. Likely we won't be able to tell which is best until we can run this in constraints.

File: testdata/zkc/bench/keccakf_with_padding.zkc (hunk 5 above)
Comment (OlivierBBB) [reply testdata/zkc/bench/keccakf_with_padding.zkc:15]: Ok, I somehow missed that.

File: testdata/zkc/bench/keccakf_le.zkc (hunk 4 above)
Comment (amkCha) [reply testdata/zkc/bench/keccakf_le.zkc:15]: As discussed during the daily, I might need to wait to have the constraints count to know which route to take here : start from a u1088 and destructure it or get u8 and re-structure them into u64 lanes
